File: admin-server/server/plugin/sms/router/sms_platform_business_types.go
package router

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

var SmsPlatformBusinessTypes = new(smsPlatformBusinessTypes)

type smsPlatformBusinessTypes struct {}

// Init 初始化 平台业务 路由信息
func (r *smsPlatformBusinessTypes) Init(public *gin.RouterGroup, private *gin.RouterGroup) {
	{
	    group := private.Group("smsPlatformBusinessTypes").Use(middleware.OperationRecord())
		group.POST("createSmsPlatformBusinessTypes", apiSmsPlatformBusinessTypes.CreateSmsPlatformBusinessTypes)   // 新建平台业务
		group.DELETE("deleteSmsPlatformBusinessTypes", apiSmsPlatformBusinessTypes.DeleteSmsPlatformBusinessTypes) // 删除平台业务
		group.DELETE("deleteSmsPlatformBusinessTypesByIds", apiSmsPlatformBusinessTypes.DeleteSmsPlatformBusinessTypesByIds) // 批量删除平台业务
		group.PUT("updateSmsPlatformBusinessTypes", apiSmsPlatformBusinessTypes.UpdateSmsPlatformBusinessTypes)    // 更新平台业务
	}
	{
	    group := private.Group("smsPlatformBusinessTypes")
		group.GET("findSmsPlatformBusinessTypes", apiSmsPlatformBusinessTypes.FindSmsPlatformBusinessTypes)        // 根据ID获取平台业务
		group.GET("getSmsPlatformBusinessTypesList", apiSmsPlatformBusinessTypes.GetSmsPlatformBusinessTypesList)  // 获取平台业务列表
	}
	{
	    group := public.Group("smsPlatformBusinessTypes")
	    group.GET("getSmsPlatformBusinessTypesPublic", apiSmsPlatformBusinessTypes.GetSmsPlatformBusinessTypesPublic)  // 平台业务开放接口
	}
}

File: admin-server/web/src/plugin/sms/api/smsPlatformBusinessTypes.js
import service from '@/utils/request'
// @Tags SmsPlatformBusinessTypes
// @Summary 创建平台业务
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body model.SmsPlatformBusinessTypes true "创建平台业务"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"创建成功"}"
// @Router /smsPlatformBusinessTypes/createSmsPlatformBusinessTypes [post]
export const createSmsPlatformBusinessTypes = (data) => {
  return service({
    url: '/smsPlatformBusinessTypes/createSmsPlatformBusinessTypes',
    method: 'post',
    data
  })
}

// @Tags SmsPlatformBusinessTypes
// @Summary 删除平台业务
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body model.SmsPlatformBusinessTypes true "删除平台业务"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /smsPlatformBusinessTypes/deleteSmsPlatformBusinessTypes [delete]
export const deleteSmsPlatformBusinessTypes = (params) => {
  return service({
    url: '/smsPlatformBusinessTypes/deleteSmsPlatformBusinessTypes',
    method: 'delete',
    params
  })
}

// @Tags SmsPlatformBusinessTypes
// @Summary 批量删除平台业务
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除平台业务"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /smsPlatformBusinessTypes/deleteSmsPlatformBusinessTypes [delete]
export const deleteSmsPlatformBusinessTypesByIds = (params) => {
  return service({
    url: '/smsPlatformBusinessTypes/deleteSmsPlatformBusinessTypesByIds',
    method: 'delete',
    params
  })
}

// @Tags SmsPlatformBusinessTypes
// @Summary 更新平台业务
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body model.SmsPlatformBusinessTypes true "更新平台业务"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /smsPlatformBusinessTypes/updateSmsPlatformBusinessTypes [put]
export const updateSmsPlatformBusinessTypes = (data) => {
  return service({
    url: '/smsPlatformBusinessTypes/updateSmsPlatformBusinessTypes',
    method: 'put',
    data
  })
}

// @Tags SmsPlatformBusinessTypes
// @Summary 用id查询平台业务
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data query model.SmsPlatformBusinessTypes true "用id查询平台业务"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /smsPlatformBusinessTypes/findSmsPlatformBusinessTypes [get]
export const findSmsPlatformBusinessTypes = (params) => {
  return service({
    url: '/smsPlatformBusinessTypes/findSmsPlatformBusinessTypes',
    method: 'get',
    params
  })
}

// @Tags SmsPlatformBusinessTypes
// @Summary 分页获取平台业务列表
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data query request.PageInfo true "分页获取平台业务列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /smsPlatformBusinessTypes/getSmsPlatformBusinessTypesList [get]
export const getSmsPlatformBusinessTypesList = (params) => {
  return service({
    url: '/smsPlatformBusinessTypes/getSmsPlatformBusinessTypesList',
    method: 'get',
    params
  })
}
// @Tags SmsPlatformBusinessTypes
// @Summary 不需要鉴权的平台业务接口
// @Accept application/json
// @Produce application/json
// @Param data query request.SmsPlatformBusinessTypesSearch true "分页获取平台业务列表"
// @Success 200 {object} response.Response{data=object,msg=string} "获取成功"
// @Router /smsPlatformBusinessTypes/getSmsPlatformBusinessTypesPublic [get]
export const getSmsPlatformBusinessTypesPublic = () => {
  return service({
    url: '/smsPlatformBusinessTypes/getSmsPlatformBusinessTypesPublic',
    method: 'get',
  })
}
